### DISCUSSION TITLE
Release v1.1.0

### DISCUSSION THREAD

**Discussion by warrensbox** [warrensbox/terraform-switcher#379] [Releases]
Notes related to Release v1.1.0

**Comment by warrensbox** [comment]
Notes from @MatrixCrawler : 
>I added some PR to the 1.1.0 Milestone. Feel free to remove some or add more.
At moment there would be 

https://github.com/warrensbox/terraform-switcher/pull/356
https://github.com/warrensbox/terraform-switcher/pull/361
https://github.com/warrensbox/terraform-switcher/pull/362

**Reply by warrensbox** [reply]
Everything that is merged to master (after 1.0.2) will be released

**Comment by warrensbox** [comment]
How useful if the information in the square brackets?

```
01:49:49.723 DEBUG [list_versions.go:19,getTFList] Get list of terraform versions
01:49:49.832 INFO [semver.go:42,SemVerParser] Matched version: "0.15.5"
01:49:49.832 INFO [list_versions.go:212,ShowLatestImplicitVersion] 0.15.5
```
The end users will have to go to the source code to debug the code. I don't think they will do that. Especially for INFO or NOTICE.
For example, in this log `01:49:49.832 INFO [semver.go:42,SemVerParser] Matched version: "0.15.5"`,  the  `[semver.go:42,SemVerParser] ` part is  not useful information to the end user.

It might be useful for DEBUG or TRACE but for INFO, it just unnecessary information, especially when the tool is installed as a binary.

**Reply by MatrixCrawler** [reply]
If there is a bug we have it a lot easier with this log information.
But we could limit that to DEBUG level. But then the user would have to run with explicit debug level for the Issue report.

**Reply by yermulnik** [reply]
Support that ⬆️ 
From the user point of view this info is superfluous in vast majority of use cases. For us it is useful in case of bug/issue reports.
Let's configure logger to have this piece added to all log messages only when `DEBUG` level of logging is explicitly requested by the user (and ask bug/issue reporters to provide details with explicit debug level).

**Reply by warrensbox** [reply]
We don't have to fix this if we don't have the time. We can do this in our next release.

**Reply by MatrixCrawler** [reply]
Added that to the PR. Was only a minor change.

**Comment by MatrixCrawler** [comment]
@warrensbox do you want to trigger the release? We have done all for the milestone.

**Reply by warrensbox** [reply]
Yes, I can do that

**Reply by warrensbox** [reply]
Looks like there's an issue with the deployment. I will try to debug it in the morning. If you have a fix for it, please go for it.

**Reply by MatrixCrawler** [reply]
I think you'd have to exempt PR's that are created by the bot from having Approvals.
On the other hand: Do we really NEED the version file? Where is it used? If we don't modify it we would not have that problem

**Reply by yermulnik** [reply]
Would be good to update CHANGELOG and cut the release afterwards.

**Reply by yermulnik** [reply]
https://github.com/warrensbox/terraform-switcher/pull/386

**Reply by yermulnik** [reply]
_(process is still mainly manual 😢 I'll search for solution/tool to help automate it)_

**Reply by yermulnik** [reply]
@warrensbox I think #386 should be merged right before releasing `v1.1.0` (so that we don't have changelog running ahead of time) — please do it when you get automation fixed and ready to cut `v1.0.0` release. Thanks.

**Reply by warrensbox** [reply]
We do not need the VERSION file anymore, we used to use it in our old way of versioning.

**Reply by warrensbox** [reply]
https://github.com/warrensbox/terraform-switcher/pull/387

- Previous failed test show that the version that would be release would be v1.0.0 instead of 1.10
- I believe that's because we appended v to the version number, I initialized the version to be 1.1.0
- I also removed the part where the release job added the version to the version file
- Also, remove version file

**Reply by warrensbox** [reply]
I have merged #386 @yermulnik

**Reply by warrensbox** [reply]
@MatrixCrawler I having issues with the deployment:

https://github.com/warrensbox/terraform-switcher/actions/runs/8824308335/job/24226509959

```
Run gh pr merge --merge --auto ""
  
no pull requests found for branch "master"
Error: Process completed with exit code 1.
```

**Reply by warrensbox** [reply]
I put a step to check the output:

```

    - name: Check outputs
      if: ${{ steps.cpr.outputs.pull-request-number }}
      run: |
        echo "Pull Request Number - ${{ steps.cpr.outputs.pull-request-number }}"
        echo "Pull Request URL - ${{ steps.cpr.outputs.pull-request-url }}"
 ```
 
 and it is not getting invoked. I am assuming that the `Create Pull Request` step is not creating a PR.

**Reply by warrensbox** [reply]
I **commented** out the automatic PR and Merge here:

https://github.com/warrensbox/terraform-switcher/blob/master/.github/workflows/release.yml

I was facing issues releasing the code with that configuration. **We can add this back before the next release if we can figure out the bug**. 

I also had to add `NO_CREATE_CHANGELOG: true` as an environment variable for gorelaser:

https://github.com/warrensbox/terraform-switcher/blob/master/.github/workflows/release.yml#L122C9-L122C34

So that it does not override @yermulnik's  changelog.

**Reply by yermulnik** [reply]
> I also had to add `NO_CREATE_CHANGELOG: true` as an environment variable for gorelaser:
> 
> [`master`/.github/workflows/release.yml#L122C9-L122C34](https://github.com/warrensbox/terraform-switcher/blob/master/.github/workflows/release.yml#L122C9-L122C34)
> 
> So that it does not override @yermulnik's changelog.

@warrensbox This Goreleaser's config section is about putting changelog list into the **release** page rather than into the main CHANGELOG (at least as far as understood it).

**Reply by yermulnik** [reply]
I enabled it back: https://github.com/warrensbox/terraform-switcher/pull/418

**Comment by warrensbox** [comment]
Version 1.1.0 is officially released: https://github.com/warrensbox/terraform-switcher/releases/tag/v1.1.0

Big shoutout to everyone who had a hand in making this release happen! Couldn't have done it without each and every one of you. Thanks a million! 

Cheers! 🍻
